jwe/internal/cmd/genheader/main.go

```diff
@@ -483,72 +495,41 @@ func generateHeaders() error {
 	fmt.Fprintf(&buf, "\nreturn nil")
 	fmt.Fprintf(&buf, "\n}")
 
-	fmt.Fprintf(&buf, "\n\nfunc (h *stdHeaders) MarshalJSON() ([]byte, error) {")
-	fmt.Fprintf(&buf, "\nh.mu.RLock()")
-	fmt.Fprintf(&buf, "\ndefer h.mu.RUnlock()")
-	fmt.Fprintf(&buf, "\nvar proxy standardHeadersMarshalProxy")
-	fmt.Fprintf(&buf, "\nif h.jwk != nil {")
-	fmt.Fprintf(&buf, "\njwkbuf, err := json.Marshal(h.jwk)")
-	fmt.Fprintf(&buf, "\nif err != nil {")
-	fmt.Fprintf(&buf, "\nreturn nil, errors.Wrap(err, `failed to marshal jwk field`)")
-	fmt.Fprintf(&buf, "\n}")
-	fmt.Fprintf(&buf, "\nproxy.Xjwk = jwkbuf")
-	fmt.Fprintf(&buf, "\n}")
-
-	fmt.Fprintf(&buf, "\nif h.ephemeralPublicKey != nil {")
-	fmt.Fprintf(&buf, "\nepkbuf, err := json.Marshal(h.ephemeralPublicKey)")
-	fmt.Fprintf(&buf, "\nif err != nil {")
-	fmt.Fprintf(&buf, "\nreturn nil, errors.Wrap(err, `failed to marshal epk field`)")
-	fmt.Fprintf(&buf, "\n}")
-	fmt.Fprintf(&buf, "\nproxy.XephemeralPublicKey = epkbuf")
+	fmt.Fprintf(&buf, "\n\nfunc (h stdHeaders) MarshalJSON() ([]byte, error) {")
+	fmt.Fprintf(&buf, "\nctx, cancel := context.WithCancel(context.Background())")
+	fmt.Fprintf(&buf, "\ndefer cancel()")
+	fmt.Fprintf(&buf, "\ndata := make(map[string]interface{})")
+	fmt.Fprintf(&buf, "\nfields := make([]string, 0, %d)", len(fields))
+	fmt.Fprintf(&buf, "\nfor iter := h.Iterate(ctx); iter.Next(ctx); {")
+	fmt.Fprintf(&buf, "\npair := iter.Pair()")
+	fmt.Fprintf(&buf, "\nfields = append(fields, pair.Key.(string))")
+	fmt.Fprintf(&buf, "\ndata[pair.Key.(string)] = pair.Value")
```

Comment:
This receiver change looks correct, but is warned by go vet.
I think it's better if there are any comments left.

```
# github.com/lestrrat-go/jwx/jwe
jwe/headers_gen.go:640:9: MarshalJSON passes lock by value: github.com/lestrrat-go/jwx/jwe.stdHeaders contains sync.RWMutex
```

Reply:
Since we're doing `NewHeader()`, I think it's safe to just use a pointer

```diff
diff --git a/jwe/headers_gen.go b/jwe/headers_gen.go
index 222022c..461add5 100644
--- a/jwe/headers_gen.go
+++ b/jwe/headers_gen.go
@@ -89,7 +89,7 @@ type stdHeaders struct {
        x509CertThumbprintS256 *string                         //
        x509URL                *string                         //
        privateParams          map[string]interface{}
-       mu                     sync.RWMutex
+       mu                     *sync.RWMutex
 }
 
 type standardHeadersMarshalProxy struct {
@@ -113,6 +113,7 @@ type standardHeadersMarshalProxy struct {
 
 func NewHeaders() Headers {
        return &stdHeaders{
+               mu:            &sync.RWMutex{},
                privateParams: map[string]interface{}{},
        }
 }
diff --git a/jwe/internal/cmd/genheader/main.go b/jwe/internal/cmd/genheader/main.go
index d483d99..6b685f2 100644
--- a/jwe/internal/cmd/genheader/main.go
+++ b/jwe/internal/cmd/genheader/main.go
@@ -282,7 +282,7 @@ func generateHeaders() error {
                fmt.Fprintf(&buf, "\n%s %s // %s", f.name, fieldStorageType(f.typ), f.comment)
        }
        fmt.Fprintf(&buf, "\nprivateParams map[string]interface{}")
-       fmt.Fprintf(&buf, "\nmu sync.RWMutex")
+       fmt.Fprintf(&buf, "\nmu *sync.RWMutex")
        fmt.Fprintf(&buf, "\n}") // end type StandardHeaders
 
        // Proxy is used when unmarshaling headers
@@ -303,6 +303,7 @@ func generateHeaders() error {
 
        fmt.Fprintf(&buf, "\n\nfunc NewHeaders() Headers {")
        fmt.Fprintf(&buf, "\nreturn &stdHeaders{")
+       fmt.Fprintf(&buf, "\nmu: &sync.RWMutex{},")
        fmt.Fprintf(&buf, "\nprivateParams: map[string]interface{}{},")
        fmt.Fprintf(&buf, "\n}")
        fmt.Fprintf(&buf, "\n}")
```

Reply:
I added an explicit `go vet` in lint action, and changed the mutex to a pointer